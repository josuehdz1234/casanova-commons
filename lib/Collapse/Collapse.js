"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _react = _interopRequireDefault(require("react"));
var _propTypes = _interopRequireDefault(require("prop-types"));
var _classnames = _interopRequireDefault(require("classnames"));
var _Icons = require("../Icons");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function getIcon(show, isNew) {
  if (show) {
    return _Icons.ChevronDown;
  }
  if (!isNew) {
    return _Icons.ChevronRight;
  }
  return _Icons.CirclePlus;
}
function Collapse(props) {
  var children = props.children,
    className = props.className,
    disabled = props.disabled,
    name = props.name,
    _onClick = props.onClick,
    show = props.show,
    Title = props.title,
    titleProps = props.titleProps,
    isNew = props.isNew;
  var Icon = getIcon(show, isNew);
  var headerClassName = (0, _classnames["default"])(show && 'show', isNew && 'disabled');
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: "card ".concat(className)
  }, /*#__PURE__*/_react["default"].createElement("div", {
    onClick: function onClick() {
      if (!disabled) {
        _onClick(name);
      }
    },
    className: "card-header ".concat(headerClassName)
  }, /*#__PURE__*/_react["default"].createElement("h5", {
    className: "mb-0"
  }, typeof Title === 'string' ? Title : /*#__PURE__*/_react["default"].createElement(Title, titleProps)), /*#__PURE__*/_react["default"].createElement("span", {
    className: "card-header-icon"
  }, disabled ? '' : /*#__PURE__*/_react["default"].createElement(Icon, null))), /*#__PURE__*/_react["default"].createElement("div", {
    className: "collapse ".concat(show ? 'show' : '')
  }, /*#__PURE__*/_react["default"].createElement("div", {
    className: "card-body"
  }, children)));
}
Collapse.propTypes = {
  show: _propTypes["default"].bool,
  disabled: _propTypes["default"].bool,
  className: _propTypes["default"].string
};
Collapse.defaultProps = {
  show: false,
  disabled: false,
  className: ''
};
var _default = Collapse;
exports["default"] = _default;