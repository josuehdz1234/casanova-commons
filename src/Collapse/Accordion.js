import React, { useState, useCallback, useEffect } from 'react';
import PropTypes from 'prop-types';

function Accordion({ children, active = null }) {
  const [current, setCurrent] = useState(active);

  const handleToggle = useCallback(
    (name) => {
      if (current === name) {
        setCurrent(null);
      } else {
        setCurrent(name);
      }
    },
    [current]
  );

  const handleRequestClose = useCallback(
    (name) => {
      if (current === name) {
        setCurrent(null);
      }
    },
    [current]
  );

  useEffect(() => {
    setCurrent(active);
  }, [active]);

  return React.Children.map(children, (child) => {
    const { name } = child.props;
    const isCurrent = current === name;

    return React.cloneElement(child, {
      show: isCurrent,
      onClick: handleToggle,
      onRequestClose: handleRequestClose,
      forceClose: handleToggle,
    });
  });
}

Accordion.propTypes = {
  children: PropTypes.node.isRequired,
  initialActive: PropTypes.oneOfType([PropTypes.string]),
};

export default Accordion;
