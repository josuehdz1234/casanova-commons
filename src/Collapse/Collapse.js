import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { ChevronDown, ChevronRight, CirclePlus } from '../Icons';

function getIcon(show, isNew) {
  if (show) {
    return ChevronDown;
  }
  if (!isNew) {
    return ChevronRight;
  }
  return CirclePlus;
}
function Collapse(props) {
  const {
    children,
    className,
    disabled,
    name,
    onClick,
    show,
    title: Title,
    titleProps,
    isNew,
  } = props;
  const Icon = getIcon(show, isNew);
  const headerClassName = classnames(show && 'show', isNew && 'disabled');

  return (
    <div className={`card ${className}`}>
      <div
        onClick={() => {
          if (!disabled) {
            onClick(name);
          }
        }}
        className={`card-header ${headerClassName}`}
      >
        <h5 className="mb-0">
          {typeof Title === 'string' ? Title : <Title {...titleProps} />}
        </h5>
        <span className="card-header-icon">{disabled ? '' : <Icon />}</span>
      </div>
      <div className={`collapse ${show ? 'show' : ''}`}>
        <div className="card-body">{children}</div>
      </div>
    </div>
  );
}

Collapse.propTypes = {
  show: PropTypes.bool,
  disabled: PropTypes.bool,
  className: PropTypes.string,
};

Collapse.defaultProps = {
  show: false,
  disabled: false,
  className: '',
};

export default Collapse;
