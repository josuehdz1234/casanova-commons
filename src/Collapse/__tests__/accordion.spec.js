import React from 'react';
import { mount } from 'enzyme';
import { render, fireEvent } from '@testing-library/react';

import Accordion from '../Accordion';
import Collapse from '../Collapse';

const DumbComponent = ({ onRequestClose, name }) => (
  <div id={`dumb-${name}`} onClick={() => onRequestClose(name)}>
    Element div
  </div>
);

describe('Accordion', () => {
  test('should render without crash', () => {
    const wrapper = render(
      <Accordion label="Fecha" name="name">
        <Collapse
          name="collapse-element"
          title="Colapse element"
          titleProps={{ prop1: 'prop1' }}
          show
          onClick={(e) => console.log(e)}
          disabled={false}
          isNew={false}
        >
          <DumbComponent />
        </Collapse>
      </Accordion>
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should close when click on another element', () => {
    const { container } = render(
      <Accordion label="Fecha" name="name">
        <Collapse
          name="collapse-element"
          title="Colapse element"
          titleProps={{ prop1: 'prop1' }}
          show
          onClick={(e) => console.log(e)}
          disabled={false}
          isNew={false}
        >
          <DumbComponent />
        </Collapse>
        <Collapse
          name="collapse-element-2"
          title="Colapse element-2"
          titleProps={{ prop1: 'prop1' }}
          show={false}
          onClick={(e) => console.log(e)}
          disabled={false}
          isNew={false}
        >
          <DumbComponent />
        </Collapse>
      </Accordion>
    );

    const headers = container.querySelectorAll('.card-header');
    fireEvent.click(headers[1]);

    const elements = container.querySelectorAll('.card');

    expect(
      elements[1].querySelector('.collapse').className.includes('show')
    ).toEqual(true);
  });

  test('should close when click on the same element', () => {
    const { container } = render(
      <Accordion label="Fecha" name="name">
        <Collapse
          name="collapse-element"
          title="Colapse element"
          titleProps={{ prop1: 'prop1' }}
          show
          onClick={(e) => console.log(e)}
          disabled={false}
          isNew={false}
        >
          <DumbComponent />
        </Collapse>
        <Collapse
          name="collapse-element-2"
          title="Colapse element-2"
          titleProps={{ prop1: 'prop1' }}
          show={false}
          onClick={(e) => console.log(e)}
          disabled={false}
          isNew={false}
        >
          <DumbComponent />
        </Collapse>
      </Accordion>
    );

    const headers = container.querySelectorAll('.card-header');
    fireEvent.click(headers[0]);
    fireEvent.click(headers[0]);

    const elements = container.querySelectorAll('.card');

    expect(
      elements[0].querySelector('.collapse').className.includes('show')
    ).toEqual(false);
  });
});
