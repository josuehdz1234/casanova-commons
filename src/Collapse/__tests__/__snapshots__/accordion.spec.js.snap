// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion should render without crash 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card "
      >
        <div
          class="card-header "
        >
          <h5
            class="mb-0"
          >
            Colapse element
          </h5>
          <span
            class="card-header-icon"
          >
            <svg
              height="100%"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 12c0 .327-.121.652-.364.901l-9.509 9.726a1.228 1.228 0 01-1.762 0 1.294 1.294 0 010-1.802L14.993 12 6.365 3.175a1.294 1.294 0 010-1.802 1.228 1.228 0 011.762 0l9.509 9.727c.243.249.364.575.364.901z"
                fill="#007AFF"
                fill-rule="nonzero"
              />
            </svg>
          </span>
        </div>
        <div
          class="collapse "
        >
          <div
            class="card-body"
          >
            <div
              id="dumb-undefined"
            >
              Element div
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card "
    >
      <div
        class="card-header "
      >
        <h5
          class="mb-0"
        >
          Colapse element
        </h5>
        <span
          class="card-header-icon"
        >
          <svg
            height="100%"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 12c0 .327-.121.652-.364.901l-9.509 9.726a1.228 1.228 0 01-1.762 0 1.294 1.294 0 010-1.802L14.993 12 6.365 3.175a1.294 1.294 0 010-1.802 1.228 1.228 0 011.762 0l9.509 9.727c.243.249.364.575.364.901z"
              fill="#007AFF"
              fill-rule="nonzero"
            />
          </svg>
        </span>
      </div>
      <div
        class="collapse "
      >
        <div
          class="card-body"
        >
          <div
            id="dumb-undefined"
          >
            Element div
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
